### PR TITLE
Understanding Errors: Slightly improve lesson and move articles to assignment

### PR DESCRIPTION
**Pull request by nikitarevenco**
## Because

I noticed that some of the links could be better fit for the assignment.

## This PR

- Moves some articles to the assignment
- Explains that errors are your friend, not foe
- Adds line breaks to improve readability

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: foundations/javascript_basics/understanding_errors.md [hunk 1]
```diff
@@ -112,27 +118,36 @@ const message = str1.push(str2);
...
+
+While you should address these warnings if possible and as soon as possible, warnings are not as significant as errors and are more likely to be informational. Warnings are typically shown in yellow, while errors are typically shown in red. Though these colors are not a rule, frequently there will be a visual differentiation between the two, regardless of the platform you are encountering them on.
 
 ### Assignment
 
 <div class="lesson-content__panel" markdown="1">
 
+1. Now, it's time to go through the documentation! Learn more about [the ReferenceError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError), the [SyntaxError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError) and the [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) from the MDN Docs.
```
Comment (MaoShizhong) [inline foundations/javascript_basics/understanding_errors.md:146]:
```suggestion
1. Now, it's time to go through the documentation! Learn more about the [ReferenceError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError), the [SyntaxError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError) and the [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) from the MDN Docs.
```

File: foundations/javascript_basics/understanding_errors.md [hunk 2]
```diff
@@ -112,27 +118,36 @@ const message = str1.push(str2);
...
+1. Go through the [tutorial on the Chrome Debugger](https://developer.chrome.com/docs/devtools/javascript/).
+<div class="lesson-note" markdown="1">
+In point 4 of step 3 of the tutorial, devtools will pause on the second line rather than at the declaration of the function. Don't worry this is expected.
+</div>
```
Comment (MaoShizhong) [inline foundations/javascript_basics/understanding_errors.md:150]:
```suggestion
1. Go through the [tutorial on the Chrome Debugger](https://developer.chrome.com/docs/devtools/javascript/).

<div class="lesson-note" markdown="1">
In point 4 of step 3 of the tutorial, devtools will pause on the second line rather than at the declaration of the function. Don't worry this is expected.
</div>

```
Blank lines because the linter sometimes misses some things like list issues if HTML tags are directly next to them.

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Just before Manon gets to this more properly, just a little nit and linting thing. Would you also be able to fix the other linting issues e.g. trailing spaces please?

**Comment by nikitarevenco** [comment]
> Just before Manon gets to this more properly, just a little nit and linting thing. Would you also be able to fix the other linting issues e.g. trailing spaces please?

For sure!

File: foundations/javascript_basics/understanding_errors.md [hunk 3]
```diff
@@ -112,41 +118,53 @@ const message = str1.push(str2);
...
 
 <div class="lesson-content__panel" markdown="1">
 
-1. Work through [this lesson on MDN](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/First_steps/What_went_wrong). Be sure to download their starter code that has intentional errors.
+1. Now, it's time to go through the documentation! Learn more about the [ReferenceError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ReferenceError), the [SyntaxError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError) and the [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) from the MDN Docs.
+1. Go through the [tutorial on the Chrome Debugger](https://developer.chrome.com/docs/devtools/javascript/).
+
+<div class="lesson-note" markdown="1">
+In point 4 of step 3 of the tutorial, devtools will pause on the second line rather than at the declaration of the function. Don't worry this is expected.
+</div>
```
Comment (MaoShizhong) [inline foundations/javascript_basics/understanding_errors.md:151]: Would you be able to put this back to how it was before, since the change is part of some work that's already assigned in #27681?

File: foundations/javascript_basics/understanding_errors.md [hunk 4]
```diff
@@ -112,41 +118,53 @@ const message = str1.push(str2);
...
 - [What is the key difference between an error and a warning?](#errors-vs-warnings)
 - [What is one method you can use to resolve an error?](#tips-for-resolving-errors)
 
 ### Additional resources
 
-This section contains helpful links to other content. It isn't required, so consider it supplemental.
+This section contains helpful links to related content. It isn't required, so consider it supplemental.
 
 - [MDN JavaScript Errors Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors)
+- There are other useful `console` methods like `console.dir` or `console.table`. A [comprehensive list of `console` methods is available at w3schools](https://www.w3schools.com/jsref/obj_console.asp).
```
Comment (MaoShizhong) [inline foundations/javascript_basics/understanding_errors.md:170]: Could you also revert this change to how it was? Again, another change that's part of already-assigned work from #27681.

File: foundations/javascript_basics/understanding_errors.md (hunk 3 above)
Comment (MaoShizhong) [inline foundations/javascript_basics/understanding_errors.md:151]: This part will be reverted anyway but just a tip for the future, something like this would be better as a nested bullet for its parent list item, as putting a note box here breaks up the list and the 3rd item ends up being the start of a new ordered list!

Comment (nikitarevenco) [reply foundations/javascript_basics/understanding_errors.md]: Sure, I have reverted it!

**Review by MaoShizhong — APPROVED** [review]
Thank you as always, @nikitarevenco!
